src/main/java/tb/common/tile/TileNodeLinker.java

```diff
@@ -152,23 +153,29 @@ public void updateEntity() {
         } else instability = 0;
     }
 
+    private static final Block STABILIZER_BLOCK = OreDictionary.doesOreNameExist("blockIchorium")
+        && !OreDictionary.getOres("blockIchorium")
+            .isEmpty() ? Block.getBlockFromItem(
+                OreDictionary.getOres("blockIchorium")
+                    .get(0)
+                    .getItem())
+                : null;
+
+    private boolean hasStabilizerBlock() {
+        return STABILIZER_BLOCK != null
+            && this.worldObj.getBlock(this.xCoord, this.yCoord + 1, this.zCoord) == STABILIZER_BLOCK;
+    }
+
     public boolean instabilityCheck() {
         if (this.worldObj.rand.nextInt(50) <= this.instability) {
             int rnd = this.worldObj.rand.nextInt(this.instability);
             if (rnd == 49) {
                 // if a block of Ichorium is above dont explode or harm node
-                if (!OreDictionary.doesOreNameExist("blockIchorium")
-                    || (OreDictionary.getOres("blockIchorium") != null && !(OreDictionary.getOres("blockIchorium")
-                        .contains(new ItemStack(this.worldObj.getBlock(this.xCoord, this.yCoord + 1, this.zCoord))))))
-                    instability -= explodeTransmitter();
+                if (!hasStabilizerBlock()) instability -= explodeTransmitter();
             } else {
                 if (rnd >= 45) {
                     // if a block of Ichorium is above dont explode or harm node
-                    if (!OreDictionary.doesOreNameExist("blockIchorium")
-                        || (OreDictionary.getOres("blockIchorium") != null && !(OreDictionary.getOres("blockIchorium")
-                            .contains(
-                                new ItemStack(this.worldObj.getBlock(this.xCoord, this.yCoord + 1, this.zCoord))))))
-                        instability -= harmTransmitter();
+                    if (!hasStabilizerBlock()) instability -= harmTransmitter();
```

Comment:
Put brackets here though, while I personally prefer doing it this way, common gtnh practice say to make it explicit

Reply:
I agree with you, I wanted to keep it as close to the original as possible.

Reply:
/shrug, it is what it is, c'est la vie